atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheImpl.java

```diff
@@ -86,18 +85,28 @@ public CommitUpdate getCommitUpdate(long startTs) {
                 "start or commit info not processed for start timestamp");
 
         CommitInfo commitInfo = maybeCommitInfo.get();
-        return eventLog.getEventsBetweenVersions(startVersion, commitInfo.commitVersion())
-                .toCommitUpdate(startVersion.get(), commitInfo);
+
+        VersionBounds versionBounds = new VersionBounds.Builder()
+                .startVersion(startVersion)
+                .endVersion(commitInfo.commitVersion())
+                .build();
+
+        return eventLog.getEventsBetweenVersions(versionBounds).toCommitUpdate(startVersion.get(), commitInfo);
     }
 
     @Override
     public TransactionsLockWatchUpdate getUpdateForTransactions(
```

Comment:
Would we get anything in terms of simplification of code if this was taking Long startTimestamp, instead of Set<Long>? atlas-proxy just needs one transaction pushed here.

Reply:
Yes, that would be simpler, but I don't think by leaps and bounds - it just simplifies what we can compress a little.